src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -523,6 +548,11 @@ void applyTo(final CRAMFileReader underlyingReader, final SamReader reader) {
             void applyTo(final SRAFileReader underlyingReader, final SamReader reader) {
                 underlyingReader.enableIndexCaching(true);
             }
+
+            @Override
+            void applyTo(final HtsgetBAMFileReader underlyingReader, final SamReader reader) {
```

Comment:
This design is so crufty.  It seems like someone started refactoring it to use ReaderImplementation so we don't need these custom handlers here as well but then never actually did it.  (Just me venting... nothing to do here...)